File: InternetBulletin.Shared/DTOs/Posts/UpdatePostDTO.cs
// *********************************************************************************
//	<copyright file="UpdatePostDTO.cs" company="Personal">
//		Copyright (c) 2025 Personal
//	</copyright>
// <summary>Update post d t o.</summary>
// *********************************************************************************

namespace InternetBulletin.Shared.DTOs.Posts
{
	/// <summary>
	/// Update post d t o.
	/// </summary>
	public class UpdatePostDTO
	{
		/// <summary>
		/// Gets or sets the post identifier.
		/// </summary>
		/// <value>
		/// The post identifier.
		/// </value>
		public Guid PostId { get; set; }

		/// <summary>
		/// Gets or sets the post title.
		/// </summary>
		/// <value>
		/// The post title.
		/// </value>
		public string PostTitle { get; set; } = string.Empty;

		/// <summary>
		/// Gets or sets the content of the post.
		/// </summary>
		/// <value>
		/// The content of the post.
		/// </value>
		public string PostContent { get; set; } = string.Empty;

		/// <summary>
		/// Gets or sets the post rating.
		/// </summary>
		/// <value>
		/// The post rating.
		/// </value>
		public int? PostRating { get; set; }

		/// <summary>
		/// Gets or sets a value indicating whether this instance is NSFW.
		/// </summary>
		/// <value>
		///   <c>true</c> if this instance is NSFW; otherwise, <c>false</c>.
		/// </value>
		public bool IsNSFW { get; set; }

		/// <summary>
		/// Gets or sets the genre tag.
		/// </summary>
		/// <value>
		/// The genre tag.
		/// </value>
		public string GenreTag { get; set; } = string.Empty;
	}
}

File: InternetBulletin.Web/Models/UpdatePostDto.js
/**
 * UpdatePostDtoModel class to represent the data for updating an existing post.
 */
class UpdatePostDtoModel {
	/**
	 * Gets or sets the post identifier.
	 * @type {string}
	 */
	postId = "";

	/**
	 * Gets or sets the post title.
	 * @type {string}
	 */
	postTitle = "";

	/**
	 * Gets or sets the content of the post.
	 * @type {string}
	 */
	postContent = "";

	/**
	 * Gets or sets the post rating.
	 * @type {number}
	 */
	postRating = 0;

	/**
	 * Gets or sets the IsNSFW flag.
	 * @type {boolean}
	 */
	isNSFW = false;

	/**
	 * Gets or sets the genre tag.
	 * @type {string}
	 */
	genreTag = "";

	/**
	 * Creates an instance of UpdatePostDtoModel.
	 * @param {string} PostId The post identifier.
	 * @param {string} PostTitle The post title.
	 * @param {string} PostContent The content of the post.
	 * @param {number} PostRating The post rating.
	 * @param {boolean} IsNSFW The NSFW rating.
	 * @param {string} GenreTag The genre tag.
	 */
	constructor(PostId, PostTitle, PostContent, PostRating, IsNSFW, GenreTag) {
		this.postId = PostId;
		this.postTitle = PostTitle;
		this.postContent = PostContent;
		this.postRating = PostRating;
		this.isNSFW = IsNSFW;
		this.genreTag = GenreTag;
	}
}

export default UpdatePostDtoModel;